### PR TITLE
use stable black release

### LINKED ISSUE #503: Use stable black release
We are still using the non-stable release of black and should change to the stable one.

### PR DESCRIPTION
**Pull request by PythonFZ**
Update `pre-commit` and `Github Actions` to use the latest version of black

### CODE REVIEW

**Review by christophlohrmann — APPROVED** [review]
Time to steal this for all my projects

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+      - name: Install package
+        run: |
+          pip install .
+      - name: Install pylint
+        run: |
+          pip install pylint perflint
+      - name: run pylint
+        continue-on-error: true
+        run: |
+          pylint mdsuite --load-plugins=perflint
```
Comment (SamTov) [inline .github/workflows/lint.yaml]: Just the github newline complaint.

File: CI/integration_tests/calculators/test_green_kubo_self_diffusion_coefficients.py [hunk 2]
```diff
@@ -99,8 +99,8 @@ def test_calculator(tmp_path):
     time_should_be = time_step * np.arange(0, vacf_range) * units.time
     thermal_vel_SI = np.sqrt(3 * kT / mass) * units.length / units.time
     relaxation_time_SI = relaxation_time * units.time
-    vacf_should_be = thermal_vel_SI ** 2 * np.exp(-time_should_be / relaxation_time_SI)
-    diff_coeff_should_be = diff_coeff * units.length ** 2 / units.time
+    vacf_should_be = thermal_vel_SI**2 * np.exp(-time_should_be / relaxation_time_SI)
```
Comment (SamTov) [inline CI/integration_tests/calculators/test_green_kubo_self_diffusion_coefficients.py:102]: Do you know the rational behind this change?

File: mdsuite/calculators/green_kubo_ionic_conductivity.py [hunk 3]
```diff
@@ -170,7 +170,7 @@ def _calculate_prefactor(self):
 
         """
         # Calculate the prefactor
-        numerator = (elementary_charge ** 2) * (self.experiment.units["length"] ** 2)
+        numerator = (elementary_charge**2) * (self.experiment.units["length"] ** 2)
```
Comment (SamTov) [inline mdsuite/calculators/green_kubo_ionic_conductivity.py:173]: And why it isn't here? Just curious

**Review by SamTov — APPROVED** [review]
Looks good, a bit confused by their change to the power rules but this is not important. There is a newline missing in the lint yaml file that github is complaining about.

File: CI/integration_tests/calculators/test_green_kubo_self_diffusion_coefficients.py (hunk 2 above)
Comment (PythonFZ) [reply CI/integration_tests/calculators/test_green_kubo_self_diffusion_coefficients.py:102]: Black changed its formating style going to the stable release with regards to `2**2` vs `2 ** 2`

File: mdsuite/calculators/green_kubo_ionic_conductivity.py (hunk 3 above)
Comment (PythonFZ) [reply mdsuite/calculators/green_kubo_ionic_conductivity.py:173]: I have no idea - funny catch. I have to admit I did not go through the changed files by black.

**Pull request merged**